popup_survey.module

```diff
@@ -73,28 +73,32 @@ function popup_survey_page_bottom(array &$page_bottom) {
     }
 
     // Randomly skip so that not all users see the popup.
-    $frequency = $frequency_override ?? $config->get('popup_survey_frequency') ?: 1;
-    if (mt_rand(1, $frequency) === 1) {
-
-      $ua = Html::escape($config->get('popup_survey_ua_exclude') ?: '');
-      $page_bottom['popup_survey'] = [
-        '#theme' => 'popup_survey',
-        '#survey_id' => $popup_config_entity_id,
-        '#cache' => [
-          'max-age' => 0,
+    $frequency = $frequency_override ?? $config->get('popup_survey_frequency') ?: 0;
+    $ua = Html::escape($config->get('popup_survey_ua_exclude') ?: '');
+    $page_bottom['popup_survey'] = [
+      '#theme' => 'popup_survey',
+      '#survey_id' => $popup_config_entity_id,
+      '#cache' => [
+        'contexts' => [
+          'url.path',
         ],
-        '#attached' => [
-          'library' => [
-            'popup_survey/popup_survey',
-          ],
-          'drupalSettings' => [
-            'popup_survey' => [
-              'botlist' => $ua,
-            ],
+        'tags' => [
+          'popup_survey_list',
```

Comment:
@minorOffense I think we're missing `config:popup_survey_list` here. Editing a single survey results in these tags in the purge queue:

- config:popup_survey.popup_survey.jaime_survey_for_varnish
- config:popup_survey_list

Reply:
Oh are they config entities? My bad, yeah that would be correct.

Reply:
Yeah the entity_type_list is for content entities, config: would be correct